src/components/JaegerIntegration/JaegerResults/SpanTable.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import { Span } from '../../../types/JaegerInfo';
+import { Table, TableHeader, TableBody, IRow, expandable, RowWrapperProps } from '@patternfly/react-table';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { Tooltip } from '@patternfly/react-core';
+import { KialiAppState } from '../../../store/Store';
+import { connect } from 'react-redux';
+import { formatDuration } from './transform';
+import history from '../../../app/History';
+import { Link } from 'react-router-dom';
+import { serverConfig } from '../../../config';
+import { SpanTabsTags } from './SpanTabsTags';
+import { isErrorTag } from '../RouteHelper';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Table/table';
+import { PfColors } from '../../Pf/PfColors';
+
+interface SpanDetailProps {
+  spans: Span[];
+  namespaceSelector: boolean;
+}
+
+interface SpanDetailState {
+  spanSelected?: Span;
+  isModalOpen: boolean;
+  columns: any;
+  rows: any;
+}
+
+export class SpanTableC extends React.Component<SpanDetailProps, SpanDetailState> {
+  constructor(props: SpanDetailProps) {
+    super(props);
+    this.state = {
+      isModalOpen: false,
+      columns: [
+        {
+          title: 'Operation',
+          cellFormatters: [expandable]
+        },
+        'Service',
+        { title: 'Duration' },
+        '',
+        ''
+      ],
+      rows: this.getRows()
+    };
+  }
+
+  componentDidUpdate(prevProps: Readonly<SpanDetailProps>): void {
+    if (prevProps.spans !== this.props.spans) {
+      this.setState({ rows: this.getRows() });
+    }
+  }
+
+  goService = (service: string = this.props.spans[0].process.serviceName, extra: string = '') => {
+    const ns = service.split('.')[1] || serverConfig.istioNamespace;
+    const srv = service.split('.')[0];
+    return '/namespaces/' + ns + '/services/' + srv + extra;
+  };
+
+  goLogsWorkloads = (workload: string, srv: string) => {
+    const ns = srv.split('.')[1] || serverConfig.istioNamespace;
+    return '/namespaces/' + ns + '/workloads/' + workload + '?tab=logs';
+  };
+
+  getNodeLog = (sp: Span) => {
+    const node = sp.tags.filter(tag => tag.key === 'node_id')[0].value;
+    const srv = sp.process.serviceName.split('.')[0];
+    const regex = new RegExp(`${srv}-v[0-9]*`);
```

Comment:
I don't this will work in all cases. It's pretty tight to "v1/v2, but one with "version=prod" in Istio will fail.

So, if node_id format is:
sidecar~10.130.2.14~productpage-v1-5fc4d7dbc9-9j6bk.bookinfo~bookinfo.svc.cluster.local

better to match something like "productpage-prod" so RegExp perhaps needs to take anything between two dashes I guess.

Reply:
mmm yeah it could be a solution, but if user create some kind of tag like v1-dev or v1-prod we'll have the same problem :(